### PR TITLE
Replace SQL COUNT call for Exists with SQL EXISTS

### PR DESCRIPTION
**Pull request by stanislas-m**
This change makes use of the SQL `EXISTS` statement, so the DB doesn't need to count all the rows to check existence.

### CODE REVIEW

File: finders.go [hunk 1]
```diff
@@ -256,8 +256,32 @@ func (q *Query) eagerAssociations(model interface{}) error {
 //
 // 	q.Where("name = ?", "mark").Exists(&User{})
 func (q *Query) Exists(model interface{}) (bool, error) {
-	i, err := q.Count(model)
-	return i != 0, err
+	tmpQuery := Q(q.Connection)
+	q.Clone(tmpQuery) //avoid mendling with original query
```
Comment (PetarAtanasov) [inline finders.go]: typo, should be   "avoid meddling with the original query"

File: finders.go [hunk 2]
```diff
@@ -256,8 +256,32 @@ func (q *Query) eagerAssociations(model interface{}) error {
...
+	var res bool
+
+	err := tmpQuery.Connection.timeFunc("Exists", func() error {
+		tmpQuery.Paginator = nil
+		tmpQuery.orderClauses = clauses{}
+		tmpQuery.limitResults = 0
+		query, args := tmpQuery.ToSQL(&Model{Value: model})
+
+		// when query contains custom selected fields / executed using RawQuery,
+		// sql may already contains limit and offset
```
Comment (PetarAtanasov) [inline finders.go]: Sorry, another typo.
Should be "contain", not plural.

File: finders.go (hunk 1 above)
Comment (stanislas-m) [reply finders.go]: That's a copy-paste from `Count`, I'll fix it there too. :)

File: finders.go (hunk 2 above)
Comment (stanislas-m) [reply finders.go]: That's not plural, `contains` refers to `query`. That's the third person form, in present tense.

**Pull request merged**